### PR TITLE
enable using filesystem from a UNC location

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
The problem is while validating the path, it splits `\\server\share\folder` by the backslash so the root is just `\` and the path ends up being `\server\share\folder` which isn't valid.  Fix is on Windows to add extra slash if the root is `\`

Fix https://github.com/PowerShell/PowerShell/issues/4990

### CODE REVIEW

File: src/System.Management.Automation/namespaces/NavigationProviderBase.cs [hunk 1]
```diff
@@ -424,6 +424,13 @@ protected string MakePath(string parent, string child, bool childIsLeaf)
                     // Joins the paths
 
                     StringBuilder builder = new StringBuilder(parent, parent.Length + child.Length + 1);
+#if !UNIX
+                    // On Windows, if we have a single backslash for the parent, it's a UNC path so we need to add another backslash back
+                    if (String.Compare(parent, StringLiterals.DefaultPathSeparatorString, StringComparison.Ordinal) == 0)
+                    {
+                        builder.Append('\\');
```
Comment (anmenaga) [inline src/System.Management.Automation/namespaces/NavigationProviderBase.cs]: Probably better to use DefaultPathSeparatorString instead of char constant.

**Review by anmenaga — CHANGES_REQUESTED** [review]

**Comment by SteveL-MSFT** [comment]
I don't like the change being outside of the filesystemprovider, let me see if I can rework this.

**Comment by daxian-dbw** [comment]
@SteveL-MSFT yeah, it would be the best if we can fix the `GetParentPath` to take into account UNC paths.

**Comment by SteveL-MSFT** [comment]
@daxian-dbw `GetParentPath` gets called repeatedly to validate the entire path from leaf to root.  I think putting the fix in NormalizePath() may be better.

File: test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1 [hunk 2]
```diff
@@ -1301,3 +1301,19 @@ Describe "Extended FileSystem Path/Location Cmdlet Provider Tests" -Tags "Featur
         }
     }
 }
+
+Describe "UNC paths" -Tags 'CI' {
+    It "Can Get-ChildItems from a UNC location" -Skip:(!$IsWindows) {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1]: Should we check Set-Location too?

**Review by iSazonov — COMMENTED** [review]
I unlike the fix - it seems a performance is bad.
I do `cd \\localhost\c$` and discover that we call `NormalizePath` 6-9 times for the same path!

File: test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1 (hunk 2 above)
Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1]: Sure

**Comment by SteveL-MSFT** [comment]
@iSazonov I've noticed that the providers get called a lot.  Perhaps GetParentPath() is the right place even thought that gets called a lot as well, but we only need to make it a UNC path again if it's at the root.

**Comment by iSazonov** [comment]
I wonder why we have to call NormalizePath method so many times on the same path? Maybe we can fix this?

**Comment by SteveL-MSFT** [comment]
@iSazonov I created https://github.com/PowerShell/PowerShell/issues/5002 as the perf work item

**Comment by adityapatwardhan** [comment]
@anmenaga  Please update your review.

**Comment by adityapatwardhan** [comment]
@anmenaga  ping...

**Review by anmenaga — APPROVED** [review]
LGTM.
Too many paths. :)

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Pull request merged**